src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -334,28 +381,31 @@ private void WorkerLoop()
                                     Debug.Assert(gotActiveOp, "Expected to find GCHandle ptr in active operations table");
                                     if (gotActiveOp)
                                     {
-                                        DeactivateActiveRequest(multiHandle, completedOperation.Easy, gcHandlePtr, completedOperation.CancellationRegistration);
+                                        DeactivateActiveRequest(completedOperation.Easy, gcHandlePtr, completedOperation.CancellationRegistration);
                                         FinishRequest(completedOperation.Easy, result);
                                     }
                                 }
                             }
                         }
 
-                        // Wait for more things to do.
-                        bool isWakeupRequestedPipeActive;
-                        bool isTimeout;
-                        ThrowIfCURLMError(Interop.Http.MultiWait(multiHandle, _wakeupRequestedPipeFd, out isWakeupRequestedPipeActive, out isTimeout));
-                        if (isWakeupRequestedPipeActive)
+                        // If there are any active operations, wait for more things to do.
+                        if (_activeOperations.Count > 0)
                         {
-                            // We woke up (at least in part) because a wake-up was requested.  
-                            // Read the data out of the pipe to clear it.
-                            Debug.Assert(!isTimeout, "should not have timed out if isExtraFileDescriptorActive");
-                            EventSourceTrace("Wait wake-up");
-                            ReadFromWakeupPipeWhenKnownToContainData();
-                        }
-                        if (isTimeout)
-                        {
-                            EventSourceTrace("Wait timeout");
+                            bool isWakeupRequestedPipeActive;
+                            bool isTimeout;
+                            ThrowIfCURLMError(Interop.Http.MultiWait(_multiHandle, _wakeupRequestedPipeFd, out isWakeupRequestedPipeActive, out isTimeout));
+                            if (isWakeupRequestedPipeActive)
+                            {
+                                // We woke up (at least in part) because a wake-up was requested.  
+                                // Read the data out of the pipe to clear it.
+                                Debug.Assert(!isTimeout, "should not have timed out if isExtraFileDescriptorActive");
```

Comment:
I know it was already this way, but the assert message and the variable don't match.